File: services/tradelog/tradelog-api/src/main/java/com/example/tradelog/api/repository/OptionJournalModelRowMapper.java
package com.example.tradelog.api.repository;

import com.example.common.converter.TimeConverter;
import com.example.tradelog.api.core.model.ActionType;
import com.example.tradelog.api.core.model.OptionJournalModel;
import com.example.tradelog.api.core.model.OptionType;
import com.example.tradelog.api.core.model.TransactionModel;
import com.example.tradelog.api.db.converter.TransactionModelRowMapper;
import org.springframework.jdbc.core.RowMapper;

import java.sql.ResultSet;
import java.sql.SQLException;

public class OptionJournalModelRowMapper implements RowMapper<OptionJournalModel> {

    @Override
    public OptionJournalModel mapRow(ResultSet rs, int rowNum) throws SQLException {
        TransactionModel transactionModel = new TransactionModelRowMapper(rs).invoke();

        return new OptionJournalModel(transactionModel,
                rs.getDouble("stock_price"),
                rs.getDouble("strike_price"),
                TimeConverter.fromTimestamp(rs.getTimestamp("expiry_date")),
                rs.getInt("contract_number"),
                rs.getDouble("premium"),
                ActionType.Companion.lookup(rs.getString("action_fk")),
                OptionType.Companion.lookup(rs.getString("option_type_fk")));
    }
}

File: services/tradelog/tradelog-api/src/main/kotlin/com/example/tradelog/api/db/converter/OptionJournalModelRowMapper.kt
package com.example.tradelog.api.db.converter

import com.example.common.converter.TimeConverter
import com.example.tradelog.api.core.model.ActionType
import com.example.tradelog.api.core.model.OptionJournalModel
import com.example.tradelog.api.core.model.OptionType
import org.springframework.jdbc.core.RowMapper
import java.sql.ResultSet

class OptionJournalModelRowMapper: RowMapper<OptionJournalModel> {

    override fun mapRow(rs: ResultSet, rowNum: Int): OptionJournalModel {
        val transactionModel = TransactionModelRowMapper(rs).invoke()

        return OptionJournalModel(
                transactionDetails = transactionModel,
                stockPrice = rs.getDouble("stock_price"),
                strikePrice = rs.getDouble("strike_price"),
                expiryDate = TimeConverter.fromTimestamp(rs.getTimestamp("expiry_date")),
                contracts = rs.getInt("contract_number"),
                premium = rs.getDouble("premium"),
                action = ActionType.lookup(rs.getString("action_fk")),
                optionType = OptionType.lookup(rs.getString("option_type_fk"))
        )
    }

}